### ISSUE TITLE
Accessibility issues in Tooltip and Dialog components

### ISSUE THREAD

**Issue by tambaqui** [primefaces/primereact#862]
**Tooltips:**
Tooltips should include the _role="tooltip"_ and an _id_. When the tooltip is opened the trigger element should include the _aria-describedby_ attribute pointing to that _id_.

**Dialogs:**
When opening a dialog, the rest of the contents of the page should be hidden via _aria-hidden="true"_, leaving only the dialog visible to screen readers. To control this aspect, it would be more appropriate for the dialogue to be added to the end of the body when it is opened and not in context.

**Comment by mertsincan** [comment]
Hi,

For Tooltip, I added id property and role="tooltip" to html element.

For Dialogs, I think this is not related to dialog element. Please see; https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/dialog.html (_Notes on aria-modal and aria-hidden_ section) 

Best Regards,

**Comment by tambaqui** [comment]
Hi @mertsincan, take a look at the "Notes on aria-modal and aria-hidden" in the section you indicate, below:

1. The aria-modal property was introduced in ARIA 1.1. As a new property, screen reader users may experience varying degrees of support for it.
2. Applying the aria-modal property to the dialog element replaces the technique of using aria-hidden on the background for informing assistive technologies that content outside a dialog is inert.
3. In legacy dialog implementations where aria-hidden is used to make content outside a dialog inert for assistive technology users, it is important that:
    1. aria-hidden is set to true on each element containing a portion of the inert layer.
    2. The dialog element is not a descendant of any element that has aria-hidden set to true.

If you don't use the aria-hidden technique, then you should use the aria-modal = "true" property.

**Comment by mertsincan** [comment]
Hi @tambaqui,

Thanks a lot for the update! Sorry for the typo in code;
// Old code;
aria-modal={this.props.mod**e**l}

// New code;
aria-modal={this.props.mod**a**l}

If the modal property is true, the aria-modal is added.

Best Regards,

**Comment by tambaqui** [comment]
Perfect, thanks
